<launch>
    <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>
    <!-- <include file="$(find turtlebot_bringup)/launch/3dsensor.launch"/> -->

    <include file="$(find turtlebot_teleop)/launch/logitech.launch"/>
    <param name="/move_base/global_costmap/inflation_layer/inflation_radius" value="0.07"/>
    <param name="/move_base/local_costmap/inflation_layer/inflation_radius" value="0.07"/>
    <param name="/move_base/controller_frequency" value="5.0"/>
    <param name="/move_base/controller_patience" value="3.0"/>
    <param name="/move_base/planner_frequency" value="1.0"/>
    <param name="/move_base/planner_patience" value="5.0"/>
    <param name="/move_base/oscillation_timeout" value="10.0"/>
    <param name="/move_base/oscillation_distance" value="0.2"/>
    <include file="$(find turtlebot_navigation)/launch/amcl_demo.launch">
        <arg name="map_file" value = "/home/robot/catkin_ws/src/CMPUT-412-C3/map.yaml"/>
    </include>

    <!-- <include file="$(find turtlebot_rviz_launchers)/launch/view_navigation.launch"/> -->

    <arg name="marker_size" default="19.9" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />
    <arg name="cam_image_topic" default="/camera/rgb/image_raw" />
    <arg name="cam_info_topic" default="/camera/rgb/camera_info" />
    <arg name="output_frame" default="/base_link" />
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size"           type="double" value="$(arg marker_size)" />
        <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
        <param name="output_frame"          type="string" value="$(arg output_frame)" />

        <remap from="camera_image"  to="$(arg cam_image_topic)" />
        <remap from="camera_info"   to="$(arg cam_info_topic)" />
    </node>

    <node pkg="c3" name="c3_main" type="c3_main.py" output="screen"/>
    <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
        <param name="device" type="string" value="/dev/video1" />
        <remap from="/image_raw" to="usb_cam/image_raw"/>
    </node>

    <!--node pkg="smach_viewer" type="smach_viewer.py" name="smach_viewer" /-->
</launch>
